### DISCUSSION TITLE
Flashing failed with T-Beam Supreme

### DISCUSSION THREAD

**Discussion by ki6uve** [markqvist/Reticulum#597] [Help & Questions]
I get this error when attempting to flash a T-Beam Supreme with this firmware: rnode_firmware_tbeam_supreme.zip using the RNode Web UI Flasher here: https://liamcottle.github.io/rnode-flasher/

Firmware flashing failed: rnode_firmware_t3s3.boot_app0 not found in firmware file!
![Screenshot 2024-10-28 124009](https://github.com/user-attachments/assets/b4b59484-1264-4856-9d6d-950426c8e6c5)

**Comment by markqvist** [comment]
As far as I know, the T-Beam Supreme is not supported by @liamcottle's web flasher yet, so you'll have to use `rnodeconf` to flash that board. Using `rnodeconf --autoinstall` and following the prompts should get you going.

**Reply by ki6uve** [reply]
Doesn't look like rnodeconf is setup to use that device either.  I get this error:

               T-Beam RNode Installer

The RNode firmware can currently be installed on T-Beam devices using the
SX1276, SX1278, SX1262 and SX1268 transceiver chips.

Important! Using RNode firmware on T-Beam devices should currently be
considered experimental. It is not intended for production or critical use.
The currently supplied firmware is provided AS-IS as a courtesey to those
who would like to experiment with it. Hit enter to continue.
---------------------------------------------------------------------------


What band is this T-Beam for?

[1] 433 MHz         (with SX1278 chip)
[2] 868/915/923 MHz (with SX1276 chip)

[3] 433 MHz         (with SX1268 chip)
[4] 868/915/923 MHz (with SX1262 chip)

? 4
'clear' is not recognized as an internal or external command,
operable program or batch file.

------------------------------------------------------------------------------
                               Installer Ready

Ok, that should be all the information we need. Please confirm the following
summary before proceeding. In the next step, the device will be flashed and
provisioned, so make sure that you are satisfied with your choices.

Serial port     : COM13
Device type     : LilyGO T-Beam 850 - 950 MHz
Platform        : ESP32
Device MCU      : Espressif Systems ESP32
Firmware file   : rnode_firmware_tbeam_sx1262.zip

------------------------------------------------------------------------------

Is the above correct? [y/N] y
[14:55:14] Checking firmware file availability...
[14:55:14] Retrieving latest version info from https://unsigned.io/firmware/latest/?v=2.1.3&variant=rnode_firmware_tbeam_sx1262.zip
[14:55:14] The selected firmware for this board is version 1.78
[14:55:14] Using existing firmware file: rnode_firmware_tbeam_sx1262.zip for version 1.78
[14:55:14] Verifying firmware integrity...
[14:55:16] Decompressing firmware...
[14:55:16] Firmware decompressed
[14:55:16] Flashing RNode firmware to device on COM13
esptool.py v3.3.3
Serial port COM13
Connecting...

A fatal error occurred: This chip is ESP32-S3 not ESP32. Wrong --chip argument?
[14:55:16] Error from flasher (2) while writing.
[14:55:16] Some boards have trouble flashing at high speeds, and you can
[14:55:16] try flashing with a lower baud rate, as in this example:
[14:55:16] rnodeconf --autoinstall --baud-flash 115200

**Reply by markqvist** [reply]
It definitely is. Maybe you need to update the `rns` package to the latest version. T-Beam Supreme is a separate board, listed in the menu as:

`[12] LilyGO T-Beam Supreme`

Selecting the old version of T-Beam will not work, since it needs to set the various configuration parameters for that specific board.

**Reply by liamcottle** [reply]
> As far as I know, the T-Beam Supreme is not supported by @liamcottle's web flasher yet

@markqvist thanks for the tag. I have updated the web flasher to support this device. Do note however, I don't have one of these devices so I can't test it...

@ki6uve if you decide to try flash and provision the T-Beam Supreme via my web flasher, please let me know if it works for you :)

**Comment by ki6uve** [comment]
Can I manually choose a Firmware file using the rnode flasher?

**Comment by ki6uve** [comment]
Is there any documentation on how to setup an autonomous repeater node with the micro reticulum?  I'm doing some testing for the FreeTAKServer team and we cannot seem to make it work.  On Oct 29, 2024, at 3:13 AM, Liam Cottle ***@***.***> wrote:﻿

As far as I know, the T-Beam Supreme is not supported by @liamcottle's web flasher yet

@markqvist thanks for the tag. I have updated the web flasher to support this device. Do note however, I don't have one of these devices so I can't test it...
@ki6uve if you decide to try flash and provision the T-Beam Supreme via my web flasher, please let me know if it works for you :)

**Comment by holiman** [comment]
Tested this today, on a T-beam. 

### `rnodeconf`

It got hung on the `Verifying firmware integrity` step (I attempted twice, let it sit for ~30 minutes). That sounds to me like a step which doesn't actually involve the device, so it seems a bit odd that it would get stuck there. Thoughts on what to investigate? 

```
------------------------------------------------------------------------------
                               Installer Ready

Ok, that should be all the information we need. Please confirm the following
summary before proceeding. In the next step, the device will be flashed and
provisioned, so make sure that you are satisfied with your choices.

Serial port     : /dev/ttyACM0
Device type     : LilyGO T-Beam Supreme 850 - 950 MHz
Platform        : ESP32
Device MCU      : Espressif Systems ESP32
Firmware file   : rnode_firmware_tbeam_supreme.zip

------------------------------------------------------------------------------

Is the above correct? [y/N] y
[08:32:19] Checking firmware file availability...
[08:32:19] Retrieving latest version info from https://unsigned.io/firmware/latest/?v=2.4.0&variant=rnode_firmware_tbeam_supreme.zip
[08:32:19] The selected firmware for this board is version 1.81
[08:32:19] Using existing firmware file: rnode_firmware_tbeam_supreme.zip for version 1.81
[08:32:19] Verifying firmware integrity...
^C
```

### webflasher

Webflasher at  https://liamcottle.github.io/rnode-flasher/ worked beautifully!
